api/dbadapters/index.ts

```diff
@@ -38,7 +38,8 @@ export interface IDbAdapter extends IDbClient {
     projectConfig?: dataform.IProjectConfig
   ): Promise<dataform.IQueryEvaluation[]>;
 
-  schemas(database: string): Promise<string[]>;
+  databases(): Promise<string[]>;
+  schemas(databases?: string | string[]): Promise<string[]>;
```

Comment:
IMO: now is the time to introduce a `Schema` type (which has `database` and `schema` fields)